### ISSUE TITLE
Nightly pytorch wheel for prerelease version 2.7 is build with C++11 ABI on, at least for CPU

### ISSUE THREAD

**Issue by jeffhataws** [pytorch/pytorch#143962]
### 🐛 Describe the bug

When we install nightly pytorch 2.6 and test it with torch-xla 2.6, it appears the CPU version is build with C++11 ABI on, causing error with torch-xla. 

```
(aws_neuron_venv) [ec2-user@ip-10-1-17-115 ~]$ pip install --force-reinstall --no-deps torch --index-url https://download.pytorch.org/whl/nightly/cpu
Looking in indexes: https://download.pytorch.org/whl/nightly/cpu
Collecting torch
  Using cached https://download.pytorch.org/whl/nightly/cpu/torch-2.6.0.dev20241229%2Bcpu-cp39-cp39-manylinux_2_28_x86_64.whl (175.1 MB)
Installing collected packages: torch
  Attempting uninstall: torch
    Found existing installation: torch 2.6.0.dev20241229+cpu
    Uninstalling torch-2.6.0.dev20241229+cpu:
      Successfully uninstalled torch-2.6.0.dev20241229+cpu
Successfully installed torch-2.6.0.dev20241229+cpu
```
```
(aws_neuron_venv) [ec2-user@ip-10-1-17-115 ~]$ python3 -c "import torch_xla"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/home/ec2-user/aws_neuron_venv/lib64/python3.9/site-packages/torch_xla/__init__.py", line 20, in <module>
    import _XLAC
ImportError: /home/ec2-user/aws_neuron_venv/lib64/python3.9/site-packages/_XLAC.cpython-39-x86_64-linux-gnu.so: undefined symbol: _ZN5torch4lazy13MetricFnValueEd
```

When checking using torch.compiled_with_cxx11_abi(), we see that it returns True:
```
(aws_neuron_venv) [ec2-user@ip-10-1-17-115 ~]$ python3 -c "import torch; print(torch.compiled_with_cxx11_abi())"
True
```
The default should be with C++11 ABI off, because there are C++11 ABI wheels, for example [torch-2.6.0.dev20241229+cpu.cxx11.abi-cp310-cp310-linux_x86_64.whl](https://download.pytorch.org/whl/nightly/cpu-cxx11-abi/torch-2.6.0.dev20241229%2Bcpu.cxx11.abi-cp310-cp310-linux_x86_64.whl)

### Versions

Collecting environment information...
PyTorch version: 2.6.0.dev20241229+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Amazon Linux 2023.6.20241212 (x86_64)
GCC version: (GCC) 11.4.1 20230605 (Red Hat 11.4.1-2)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.9.20 (main, Dec 11 2024, 00:00:00)  [GCC 11.4.1 20230605 (Red Hat 11.4.1-2)] (64-bit runtime)
Python platform: Linux-6.1.119-129.201.amzn2023.x86_64-x86_64-with-glibc2.34
Is CUDA available: False
CUDA runtime version: No CUDA
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

CPU:
Architecture:                         x86_64
CPU op-mode(s):                       32-bit, 64-bit
Address sizes:                        46 bits physical, 48 bits virtual
Byte Order:                           Little Endian
CPU(s):                               8
On-line CPU(s) list:                  0-7
Vendor ID:                            GenuineIntel
Model name:                           Intel(R) Xeon(R) Platinum 8375C CPU @ 2.90GHz
CPU family:                           6
Model:                                106
Thread(s) per core:                   2
Core(s) per socket:                   4
Socket(s):                            1
Stepping:                             6
BogoMIPS:                             5799.85

Flags:                                fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ss ht syscall nx pdpe1gb rdtscp lm constant_tsc rep_good nopl xtopology nonstop_tsc cpuid aperfmperf tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm abm 3dnowprefetch invpcid_single ssbd ibrs ibpb stibp ibrs_enhanced fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves wbnoinvd ida arat avx512vbmi pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg tme avx512_vpopcntdq rdpid md_clear flush_l1d arch_capabilities
Hypervisor vendor:                    KVM
Virtualization type:                  full
L1d cache:                            192 KiB (4 instances)
L1i cache:                            128 KiB (4 instances)
L2 cache:                             5 MiB (4 instances)
L3 cache:                             54 MiB (1 instance)
NUMA node(s):                         1
NUMA node0 CPU(s):                    0-7
Vulnerability Gather data sampling:   Unknown: Dependent on hypervisor status
Vulnerability Itlb multihit:          Not affected
Vulnerability L1tf:                   Not affected
Vulnerability Mds:                    Not affected
Vulnerability Meltdown:               Not affected
Vulnerability Mmio stale data:        Mitigation; Clear CPU buffers; SMT Host state unknown
Vulnerability Reg file data sampling: Not affected
Vulnerability Retbleed:               Not affected
Vulnerability Spec rstack overflow:   Not affected
Vulnerability Spec store bypass:      Mitigation; Speculative Store Bypass disabled via prctl
Vulnerability Spectre v1:             Mitigation; usercopy/swapgs barriers and __user pointer sanitization
Vulnerability Spectre v2:             Mitigation; Enhanced / Automatic IBRS; IBPB conditional; RSB filling; PBRSB-eIBRS SW sequence; BHI SW loop, KVM SW loop
Vulnerability Srbds:                  Not affected
Vulnerability Tsx async abort:        Not affected

Versions of relevant libraries:
[pip3] numpy==1.25.2
[pip3] torch==2.6.0.dev20241229+cpu
[pip3] torch-neuronx==2.6.0.2.4.0
[pip3] torch-xla==2.6.0+git7dd2697
[pip3] torchvision==0.22.0.dev20241229+cpu
[conda] Could not collect

cc @ezyang @gchanan @zou3519 @kadeng @msaroufim @seemethere @malfet @osalpekar @atalman

**Comment by huiyan2021** [comment]
Hi @jeffhataws , you can installl torch-xla built with C++11 ABI following the guide here: https://github.com/pytorch/xla/?tab=readme-ov-file#use-nightly-build-with-c11-abi-after-10282024

**Comment by malfet** [comment]
Thank you for flagging, we should not have torch-cpu build with C++11 ABI in 2.6 yet (only cuda-12.6 binaries should be like that)

**Comment by jeffhataws** [comment]
> Thank you for flagging, we should not have torch-cpu build with C++11 ABI in 2.6 yet (only cuda-12.6 binaries should be like that)

Just curious, will torch-cpu switch to C++11 ABI default in the future? For us to plan.

**Comment by kit1980** [comment]
Removing 2.6 milestone as this is expected for the release.

**Comment by seemethere** [comment]
> > Thank you for flagging, we should not have torch-cpu build with C++11 ABI in 2.6 yet (only cuda-12.6 binaries should be like that)
> 
> Just curious, will torch-cpu switch to C++11 ABI default in the future? For us to plan.

Yes this will be the case as far as I know. cc @atalman for the most up to date news here

**Comment by atalman** [comment]
@jeffhataws Current nightly does use C++11 ABI on. And its prerelease version 2.7.

**Comment by malfet** [comment]
Not sure what have happened here, but original statement were that 2.6 prerelease binaries are compiled with CXX-11ABI, which were the reason for high pri. If this is not the case (and there are binary tests in place) then IMO it could be closed

**Comment by albanD** [comment]
Closing as expected behavior then as we do expect 2.7 to be cxx11 ABI by default indeed.

**Comment by jeffhataws** [comment]
Fixing the title as the original issue was with 2.6 where C++11 ABI should be off by default. Will check if it is fixed. Thanks.

**Comment by jeffhataws** [comment]
Confirm that the issue is fixed for PT2.6. Thanks.

**Comment by snadampal** [comment]
Hi @atalman , looks like cxx11 ABI is turned off in torch 2.6 only for the x86 wheels,  I see cxx11abi true for  aarch64 wheels 

on AWS Graviton instance:
```
~$ pip3 show torch
Name: torch
Version: 2.6.0
Summary: Tensors and Dynamic neural networks in Python with strong GPU acceleration
Home-page: https://pytorch.org/
Author: PyTorch Team
Author-email: packages@pytorch.org
License: BSD-3-Clause
Location: /home/ubuntu/.local/lib/python3.10/site-packages
Requires: filelock, fsspec, jinja2, networkx, sympy, typing-extensions
Required-by: 
```
```
~$ python3 -c "import torch; print(torch.compiled_with_cxx11_abi())"
True
```

**Comment by kit1980** [comment]
@snadampal cxx11abi true for aarch64 is expected, please see https://pytorch.org/blog/pytorch2-6/
